torchtext/experimental/transforms.py

```diff
@@ -422,3 +422,53 @@ def to_ivalue(self):
             vector = self.vector.to_ivalue()
             return VectorTransform(vector)
         return self
+
+
+class Pad(nn.Module):
+    r"""Pad sequences. Add padding id to the end of a sequences such that the sequences will
+        have the same length.
+
+    Args:
+        pad_id: the id for padding token
+        return_key_padding_mask: flag if to output padding_mask as output (Default: True)
+
+    Example:
+        >>> pad_id = 2
+        >>> pad_transform = Pad(pad_id)
+        >>> seq_batch = [torch.tensor([5, 4, 5, 6, 7]), torch.tensor([1, 3]), torch.tensor([7, 5, 8])]
+        >>> pad_seq, padding_mask = pad_transform(seq_batch)
+        >>> jit_pad_transform = torch.jit.script(pad_transform)
+    """
+
+    def __init__(self, pad_id, return_key_padding_mask=True):
+        super(Pad, self).__init__()
+        self.pad_id = pad_id
+        self.return_key_padding_mask = return_key_padding_mask
+
+    @torch.jit.export
+    def forward(self, seq_batch: List[Tensor]) -> Tuple[torch.Tensor, Optional[Tensor]]:
+        r"""Pad a list of tensors in the dim of (seq_dim, ...). The individual tensor has different length
+        (i.e. seq_dim) such that the padding function will add padding id to the end of list for the same length. It
+        assumes that the dimensions after seq_dim of the tensors are same. And the tensors have same dtype, which is
+        the dtype of output padded tensor.
+
+        Args:
+            seq_batch: a list of torch.tensor. Type: List[Tensor]
```

Comment:
Why the name seq_batch? Can we expand support to n-dim Tensors? [DE;TR has generic padding+masking code that can be used for Tensors of any dimension.](https://github.com/facebookresearch/detr/blob/master/util/misc.py#L315) Maybe that's more broadly applicable and could be used in more circumstances, maybe it's overengineering.

Reply:
If this is meant to work on only 1-dim Tensors, check the entires and throw an error. We can add support for n-dim Tensors later on.

Reply:
Here are a review of the padding functions with the JIT support in pytext library: [pad_2d](https://github.com/facebookresearch/pytext/blob/master/pytext/torchscript/utils.py#L181-L188), [pad_2d_float](https://github.com/facebookresearch/pytext/blob/8472355cb10293c833f603a461915a702311084d/pytext/torchscript/utils.py#L192-L199), [pad_2d_mask](https://github.com/facebookresearch/pytext/blob/8472355cb10293c833f603a461915a702311084d/pytext/torchscript/utils.py#L76-L107). For now, I think we can limit this PR to 1-dim tensors.